src/wing_geometry.jl

```diff
@@ -1,37 +1,46 @@
 
 """
-    mutable struct Section
+    @with_kw mutable struct Section
 
 Represents a wing section with leading edge, trailing edge, and aerodynamic properties.
 
 # Fields
 - `LE_point::MVec3` = zeros(MVec3): Leading edge point coordinates
 - `TE_point::MVec3` = zeros(MVec3): Trailing edge point coordinates
 - `aero_model`::AeroModel = INVISCID: [AeroModel](@ref)
-- `aero_data`::AeroData = nothing: See: [AeroData]    
+- `aero_data`::AeroData = nothing: See: [AeroData](@ref)
 """
 @with_kw mutable struct Section
     LE_point::MVec3 = zeros(MVec3)
     TE_point::MVec3 = zeros(MVec3)
     aero_model::AeroModel = INVISCID
     aero_data::AeroData = nothing
 end
+
 """
-    Section(LE_point::Vector{Float64}, TE_point::Vector{Float64}, 
-            aero_model=INVISCID, aero_data=nothing)
+    Section(LE_point, TE_point, aero_model)
+
+Create a new wing section with the specified leading edge point, trailing edge point, 
+and aerodynamic model.
+
+# Arguments
+- `LE_point::MVec3`: Leading edge point coordinates
+- `TE_point::MVec3`: Trailing edge point coordinates  
+- `aero_model::AeroModel`: Aerodynamic model type (e.g., INVISCID, POLAR_VECTORS)
 
-Constructor for [Section](@ref) that allows to pass Vectors of Float64 as point coordinates.
+# Returns
+- `Section`: A new section with the specified parameters and no aerodynamic data
 """
-function Section(LE_point::Vector{Float64}, TE_point::Vector{Float64}, aero_model=INVISCID, aero_data=nothing)
-    Section(MVec3(LE_point), MVec3(TE_point), aero_model, aero_data)
+function Section(LE_point, TE_point, aero_model)
+    return Section(LE_point, TE_point, aero_model, nothing)
 end
 
 """
     init!(section::Section, LE_point, TE_point, aero_model=nothing, aero_data=nothing)
 
 Function to update a [Section](@ref) in place.
 """
-function init!(section::Section, LE_point, TE_point, aero_model=nothing, aero_data=nothing)
+function init!(section::Section, LE_point::AbstractVector, TE_point::AbstractVector, aero_model=nothing, aero_data=nothing)
```

Comment:
If LE_point is a vector, perhaps write LE_points (plural). If it is a point, use the type PosVec